### PR TITLE
[Vectorized] support string type cast to bitmap

### LINKED ISSUE #8613: [Feature] support string type cast to bitmap
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

_No response_

### Use case

select k1,bitmap_to_string(e1) from baseall lateral view explode_split(bitmap_from_string('1,2,3'), '!') tmp as e1;

### Related issues

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes

Issue Number: close #8613 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/function_cast.h [hunk 1]
```diff
@@ -803,6 +804,56 @@ struct ConvertThroughParsing {
...
+//convert string to bitmap or hll type
+template <typename FromDataType, typename ToDataType, typename Name>
+struct ConvertToComplex {
+    static_assert(std::is_same_v<FromDataType, DataTypeString>,
+                  "ConvertToComplex is only applicable for String or FixedString data types");
+
+    using ToFieldType = typename ToDataType::FieldType;
+    using ColVecTo = ColumnComplexType<ToFieldType>;
+
+    template <typename Additions = void*>
```
Comment (HappenLee) [inline be/src/vec/functions/function_cast.h:816]: remove the template `Additions`

Comment (zhangstar333) [reply be/src/vec/functions/function_cast.h]: done

**Review by HappenLee — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
Why should we support converting bitmap to string?

**Comment by HappenLee** [comment]
> Why should we support converting bitmap to string?

This is to be compatible with regression testing to not cause coredump. Now directly prohibiting bitmap, HLL cast to string, which is afraid of introducing new problems.

**Review by morningman — CHANGES_REQUESTED** [review]
I think we should not allow to convert string to bitmap or vice versa, for both vec and non-vec

**Pull request closed without merging**